src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -361,6 +366,264 @@ namespace cryptonote
+        m_core.cleanup_handle_incoming_blocks(true);
+        m_core.resume_mine();
+        
+        if( bvc.m_verifivation_failed )
+        {
+          LOG_PRINT_CCONTEXT_L0("Block verification failed, dropping connection");
+          m_p2p->drop_connection(context);
+          return 1;
+        }
+        if( bvc.m_added_to_main_chain )
+        {
+          ++arg.hop;
+          //TODO: Add here announce protocol usage
+          NOTIFY_NEW_BLOCK::request reg_arg = AUTO_VAL_INIT(reg_arg);
+          reg_arg.hop = arg.hop;
+          reg_arg.current_blockchain_height = arg.current_blockchain_height;
+          reg_arg.b = b;
+          relay_block(reg_arg, context);
+        }
+        else if( bvc.m_marked_as_orphaned )
+        {
+          context.m_state = cryptonote_connection_context::state_synchronizing;
+          NOTIFY_REQUEST_CHAIN::request r = boost::value_initialized<NOTIFY_REQUEST_CHAIN::request>();
+          m_core.get_short_chain_history(r.block_ids);
+          LOG_PRINT_CCONTEXT_L2("-->>NOTIFY_REQUEST_CHAIN: m_block_ids.size()=" << r.block_ids.size() );
+          post_notify<NOTIFY_REQUEST_CHAIN>(r, context);
+        }            
+      }
+    } 
+    else
+    {
+      LOG_ERROR_CCONTEXT
+      (
+        "sent wrong block: failed to parse and validate block: \r\n"
+        << epee::string_tools::buff_to_hex_nodelimer(arg.b.block) 
+        << "\r\n dropping connection"
+      );
+        
+      m_core.resume_mine();
+      m_p2p->drop_connection(context);
+        
+      return 1;     
+    }
+        
+    return 1;
+  }  
+  //------------------------------------------------------------------------------------------------------------------------  
+  template<class t_core>
+  int t_cryptonote_protocol_handler<t_core>::handle_request_fluffy_missing_tx(int command, NOTIFY_REQUEST_FLUFFY_MISSING_TX::request& arg, cryptonote_connection_context& context)
+  {
+    LOG_PRINT_CCONTEXT_L2("NOTIFY_REQUEST_FLUFFY_MISSING_TX");
+    
+    std::list<block> local_blocks;
+    std::list<transaction> local_txs;
+    if(!m_core.get_blocks(arg.current_blockchain_height - 1, 1, local_blocks, local_txs))
+    {
+
+      LOG_ERROR_CCONTEXT
+      (
+        "Failed to handle request NOTIFY_REQUEST_FLUFFY_MISSING_TX"
+        << ", get_blocks( start_offset = " << (arg.current_blockchain_height - 1) << " ) failed"
+        << ", dropping connection"
+      );
+      
+      m_p2p->drop_connection(context);
+      return 1;            
+    }
+
+    NOTIFY_NEW_FLUFFY_BLOCK::request fluffy_response;
+    fluffy_response.b = arg.b;
+    fluffy_response.current_blockchain_height = m_core.get_current_blockchain_height();
+    fluffy_response.hop = arg.hop;    
+    size_t local_txs_count = local_txs.size();
+    BOOST_FOREACH(auto& tx_idx, arg.missing_tx_indices)
+    {
+      if(tx_idx < local_txs_count)
+      {
+        fluffy_response.b.txs.push_back(t_serializable_object_to_blob( *(std::next(local_txs.begin(), tx_idx)) ));
+      }
```

Comment:
Traced the transaction fill code to -> get_transactions(blk.tx_hashes, txs, missed_ids). The get_transaction function then fills the tx_hashes in order and pushes onto txs. So assuming, the blk.tx_hashes is in order, then the txs should be in order?